dds/DCPS/transport/framework/TransportSendBuffer.cpp

```diff
@@ -162,6 +163,7 @@ SingleSendBuffer::remove_i(BufferMap::iterator buffer_iter, BufferVec& removed)
     }
   }
 
+  destinations_.erase(buffer_iter->first);
```

Comment:
Since release_i() and remove_i() clean this up now, I think you can get rid of remove destinations_.erase(it->first) in check_capacity_i() which calls remove_i right after.